### PR TITLE
[draft] Support branching models for training

### PR DESCRIPTION
**Pull request by ragmani**
ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

**Comment by ragmani** [comment]
@nnfw-bot test onert-x64-release

File: runtime/onert/backend/train/MemoryPlanner.h [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+{
+namespace backend
+{
+namespace train
+{
+
+/**
+ * @brief Class to plan memory by bump way
+ */
+template <typename Index> class BumpPlanner : public basic::IMemoryPlanner<Index>
```
Comment (hseok-oh) [inline runtime/onert/backend/train/MemoryPlanner.h:45]: Do we need `template <typename Index>` ?

Comment (ragmani) [reply runtime/onert/backend/train/MemoryPlanner.h:45]: I guess the other temporary tensors may need another index(key), but I'm not sure it.

Comment (hseok-oh) [reply runtime/onert/backend/train/MemoryPlanner.h:45]: IMO, we can change this to use template when we need, not now.

File: runtime/onert/backend/train/BackendContext.cc [hunk 2]
```diff
@@ -44,6 +44,9 @@ backend::train::ITensorRegistry *BackendContext::genTrainingTensors()
     // NOTE Assuming there is no layout changes (Always assume NHWC or UNKNOWN)
     assert(tgraph.layout() != ir::Layout::NCHW);
 
+    if (obj.isConstant() && obj.getUses().size() > 1)
+      throw std::runtime_error("Shared constant tensor is not supported yet");
```
Comment (ragmani) [inline runtime/onert/backend/train/BackendContext.cc]: test

**Comment by ragmani** [comment]
Done.

**Pull request closed without merging**